presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/MergeJoinOptimizer.java

```diff
@@ -53,7 +55,7 @@ public PlanNode optimize(PlanNode plan, Session session, TypeProvider type, Plan
         requireNonNull(variableAllocator, "variableAllocator is null");
         requireNonNull(idAllocator, "idAllocator is null");
 
-        if (preferMergeJoin(session)) {
+        if (preferMergeJoin(session) && isGroupedExecutionEnabled(session)) {
```

Comment:
This is a partial fix. There are cases where the GroupedExecutionEnabled is set to true in the session property but grouped execution cannot be enabled. In this case, planning merge join can still lead to issues. Let's add a TODO for this to address it later.